MyCryptoHQ/MyCrypto
# Prevent invalid gas price states

Closes #888. Fixes a few erroneous gas price states from the following changes:

### Changes

* Gas slider now uses `gasPrice.value` instead of `gasPrice.raw`, preventing issues where the raw input is an invalid number, but the value is always a reliable BN.
    * We should always use `field.value` of any of these fields going forward, unless it is an `<input value={field.raw}/>`. This was a mistake on my initial pass, and we should keep an eye out for it in the future.
* Gas slider clamps its values upon mounting. Setting a very high or low gas price (e.g. 100 or 0.1) then switching back to simple view will bring the actual value inline with the slider's ranges (e.g. 100 -> 60, 0.1 -> 1)
* Gas price is removed from local storage entirely. Since it's much easier to alter per-transaction now, this was an unnecessary feature IMO.

### Steps to Test

1. Go the the send page on the `gas-price-fixes` branch.
2. Switch gas to advanced
3. Make the gas price input empty
4. Go back to simple, confirm that the gas price has been raised to one, and that there were no errors
5. Switch back to the advanced view, and set the gas price input to something high (1000)
6. Go back to simple, and confirm the gas price has been lowered to its upper bound (60)
